lib/liquid/tags/for.rb

```diff
@@ -162,19 +168,19 @@ def render_segment(context, output, segment)
         for_stack.push(loop_vars)
 
         begin
-          context['forloop'.freeze] = loop_vars
+          context['forloop'] = loop_vars
 
           segment.each do |item|
             context[@variable_name] = item
             @for_block.render_to_output_buffer(context, output)
             loop_vars.send(:increment!)
 
             # Handle any interrupts if they exist.
-            if context.interrupt?
-              interrupt = context.pop_interrupt
-              break if interrupt.is_a? BreakInterrupt
-              next if interrupt.is_a? ContinueInterrupt
-            end
+            next unless context.interrupt?
+
+            interrupt = context.pop_interrupt
+            break if interrupt.is_a?(BreakInterrupt)
+            next if interrupt.is_a?(ContinueInterrupt)
```

Comment:
Mental note for later: Dunno if it's worth it but I feel like those two `is_a?` calls could be collapsed into a single call somehow.